From: Simon McVittie <smcv@debian.org>
Date: Tue, 24 Aug 2021 21:36:26 +0100
Subject: weather: Cope with either libgweather 40 or 3.36

GNOME Shell is written in a dynamic language, so let's be dynamic.
This will let us decouple the libmutter-8-0 transition from the
libgweather transition.

Forwarded: not-needed, Debian-specific transitional code
---
 js/misc/weather.js | 23 +++++++++++++++++------
 1 file changed, 17 insertions(+), 6 deletions(-)

diff --git a/js/misc/weather.js b/js/misc/weather.js
index f617d82..cf3c82b 100644
--- a/js/misc/weather.js
+++ b/js/misc/weather.js
@@ -70,13 +70,24 @@ var WeatherClient = class {
 
         const providers =
             GWeather.Provider.METAR |
-            GWeather.Provider.MET_NO |
+            (GWeather.Provider.MET_NO ? GWeather.Provider.MET_NO : GWeather.Provider.YR_NO) |
             GWeather.Provider.OWM;
-        this._weatherInfo = new GWeather.Info({
-            application_id: 'org.gnome.Shell',
-            contact_info: 'https://gitlab.gnome.org/GNOME/gnome-shell/-/raw/master/gnome-shell.doap',
-            enabled_providers: providers,
-        });
+
+        if ((new GWeather.Info()).get_application_id) {
+            // libgweather 40
+            this._weatherInfo = new GWeather.Info({
+                application_id: 'org.gnome.Shell',
+                contact_info: 'https://gitlab.gnome.org/GNOME/gnome-shell/-/raw/master/gnome-shell.doap',
+                enabled_providers: providers,
+            });
+        }
+        else {
+            // libgweather 3.36
+            this._weatherInfo = new GWeather.Info({
+                enabled_providers: providers,
+            });
+        }
+
         this._weatherInfo.connect_after('updated', () => {
             this._lastUpdate = GLib.DateTime.new_now_local();
             this.emit('changed');
